docs/06-appearance.md.lua

```diff
@@ -0,0 +1,222 @@
+#! /usr/bin/lua
+local args = {...}
+
+local gio = require("lgi").Gio
+local gobject = require("lgi").GObject
+local glib = require("lgi").GLib
+
+local name_attr = gio.FILE_ATTRIBUTE_STANDARD_NAME
+
+-- Recursive file scanner
+local function get_all_files(path, ext, ret)
+    ret = ret or {}
+    local enumerator = gio.File.new_for_path(path):enumerate_children(
+        "FILE_ATTRIBUTE_STANDARD_NAME", 0, nil, nil
+    )
+
+    for file in function() return enumerator:next_file() end do
+        local file_name = file:get_attribute_as_string(name_attr)
+        local file_type = file:get_file_type()
+        if file_type == "REGULAR" and file_name:match(ext or "") then
+            table.insert(ret, path..file_name)
+        elseif file_type == "DIRECTORY" then
+            get_all_files(path..file_name.."/", ext, ret)
+        end
+    end
+
+    return ret
+end
+
+local function path_to_module(path)
+    for _, module in ipairs {
+        "awful", "wibox", "gears", "naughty", "menubar", "beautiful"
+    } do
+        local match = path:match("/"..module.."/([^.]+)\.lua")
+        if match then
+            return module.."."..match:gsub("/",".")
+        end
+    end
+    return ""
+end
+
+local function module_to_html(mod)
+    mod = mod:gsub("\.init", "")
+
+    local f = io.open("doc/classes/".. mod ..".html","r")
+
+    if f~=nil then
+        f:close()
+        return "../classes/".. mod ..".html"
+    end
+
+    f = io.open("doc/libraries/".. mod ..".html","r")
+
+    if f~=nil then
+        f:close()
+        return "../libraries/".. mod ..".html"
+    end
+
+    return ""
```

Comment:
Should this `error()` out instead of silently producing broken(?) output? (Same for the previous function)